mypy/typeanal.py

```diff
@@ -534,9 +563,15 @@ def visit_deleted_type(self, t: DeletedType) -> Type:
         return t
 
     def visit_type_list(self, t: TypeList) -> Type:
-        self.fail('Bracketed expression "[...]" is not valid as a type', t)
-        self.note('Did you mean "List[...]"?', t)
-        return AnyType(TypeOfAny.from_error)
+        # paramspec literal (Z[[int, str, Whatever]])
+        # TODO: invalid usage restrictions
+        params = self.analyze_callable_args(t)
+        if params:
+            ts, kinds, names = params
+            # bind these types
+            return Parameters(self.anal_array(ts), kinds, names)
+        else:
+            return AnyType(TypeOfAny.from_error)
```

Comment:
This is quite disappointing and I think there's definitely a better way to do this. (I had one previously, but it didn't work with callables, only instances...)

Reply:
Done, although the method I took broke type aliases for paramspec literals (not too big of a deal, but I'll see what can be done.)